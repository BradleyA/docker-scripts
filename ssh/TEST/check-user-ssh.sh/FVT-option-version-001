#!/bin/bash
#	ssh/TEST/check-user-ssh.sh/FVT-option-version-001  3.432.907  2019-08-11T08:42:14.043973-05:00 (CDT)  https://github.com/BradleyA/docker-security-infrastructure  uadmin  six-rpi3b.cptx86.com 3.431-3-gc04328b  
#	   updated formatting in test cases from prototype to production comments 
### production standard 3.0 shellcheck
### production standard 5.1.160 Copyright
#       Copyright (c) 2019 Bradley Allen
#       MIT License is in the online DOCUMENTATION, DOCUMENTATION URL defined below.
BOLD=$(tput -Txterm bold)
NORMAL=$(tput -Txterm sgr0)
###     Place test case here

#       TEST case Version 
SCRIPT_NAME=$(head -2 "${1}" | awk {'printf $2'})
SCRIPT_VERSION=$(head -2 "${1}" | awk {'printf $3'})
echo "${SCRIPT_NAME} ${SCRIPT_VERSION}" > ${0}.expected

"${1}" --version > "${0}".out

###
diff "${0}".expected "${0}".out >/dev/null 2>&1
RETURN_CODE=${?}
if [ ${RETURN_CODE} -eq 0 ] ; then
        echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - No difference with expected output - ${BOLD}PASS - PASS${NORMAL}"
elif [ ${RETURN_CODE} -eq 1 ] ; then
        echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - Differences with expected output - ${BOLD}FAIL - FAIL${NORMAL}"
        diff -y "${0}".expected "${0}".out
else
        echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - Test script ERROR - ${BOLD}FAIL - FAIL${NORMAL}"
fi
