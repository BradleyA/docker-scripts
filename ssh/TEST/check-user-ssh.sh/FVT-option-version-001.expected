ssh/check-user-ssh.sh 3.423.895
#	ssh/TEST/check-user-ssh.sh/FVT-option-version-001.expected  3.424.896  2019-07-28T11:03:10.169928-05:00 (CDT)  https://github.com/BradleyA/docker-security-infrastructure  uadmin  six-rpi3b.cptx86.com 3.423  
#	   testing ssh/TEST/check-user-ssh.sh/FVT-option-version-001 
